### ISSUE TITLE
Playlist no longer shows resizing cursor

### ISSUE THREAD

**Issue by low-batt** [iina/iina#4768]
**System and IINA version:**

- macOS 13.6.3
- IINA 1.3.4

**Expected behavior:**
When the cursor is positioned at the inside edge of the playlist the cursor changes from the arrow cursor to the resize-left-and-right cursor.

**Actual behavior:**
The resize-left-and-right cursor never appears when running under macOS 13.6.3. This is still working when running under macOS 10.15.7.

**Steps to reproduce:**
- Start IINA playing a video
- Click on `Show Playlist Panel` under the `Playback` menu
- Move the cursor to the inside edge of the playlist
- The cursor never switches from the arrow currsor to the resize-left-and-right cursor

- [x] MPV does not have this problem.

**How often does this happen?**
Every time.

**Comment by low-batt** [comment]
This is the code in `PlaylistView` that changes the cursor:
```swift
  override func resetCursorRects() {
    let rect = NSRect(x: frame.origin.x - 4, y: frame.origin.y, width: 4, height: frame.height)
    addCursorRect(rect, cursor: .resizeLeftRight)
  }
```
This code is working under macOS Catalina, but not under macOS Ventura

In this case I do not think this is a macOS defect. I believe that code is violating the following requirement specified in the documentation for [addCursorRect](https://developer.apple.com/documentation/appkit/nsview/1483540-addcursorrect):
> You should explicitly confine a cursor rectangle to the view's visible rectangle to prevent improper behavior.

The "- 4" puts the rectangle outside of the view's frame. I suspect this is the source of the problem.

**Comment by uiryuu** [comment]
Not directly related to this issue, but I think a range of -2 ~ 2 instead of -4 ~ 0 would be better

**Comment by low-batt** [comment]
Remember I'm not a GUI programmer, so always question my understanding of AppKit…

The suggestion is to have the resizing area span the border. That makes logical sense and might be what users expect. The downside is that it doubles the implementation required if the approach uses `resetCursorRects` as it would span views.

I've been searching the net for best practices and so far I've not found a discussion of this. My guess would be that the normal way this is handled is that the area would be fully inside the view. If you think about resizing a window you would have to have the area poke outside of the window. Is that easy to do?

Today I remembered the PR #4472 by @svobs that attempts to address the cursor issue. I don't think it was noticed that IINA has code to do this that works in macOS 10.15.7. The approach taken by this PR is probably better than what I was experimenting with. Need to check in with @svobs on this, however this issue is not as important as getting some of the other PRs updated and merged.

**Comment by svobs** [comment]
> Today I remembered the PR https://github.com/iina/iina/pull/4472 by @svobs that attempts to address the cursor issue. I don't think it was noticed that IINA has code to do this that works in macOS 10.15.7.

Yes I've found that using `NSPointInRect` is a much better approach than using a cursor rect or mouse tracking areas, which could have been the first solutions some developer scribbled on the whiteboard when the mouse was being invented. It won't work outside the window's bounds though. I haven't yet found a way to do that.

**Comment by lhc70000** [comment]
We can change this to 4.

**Comment by low-batt** [comment]
Closing this as fixed by PR #5523.
